src/chemenv/tools/converters.py

```diff
@@ -0,0 +1,387 @@
+
+import backoff
+import safe
+import deepsmiles
+import selfies
+from rdkit import Chem
+from urllib.parse import (
+    quote,
+    unquote,
+)
+from typing import Optional
+from loguru import logger
+import aiohttp
+import asyncio
+
+class Name2Smiles:
+    """Convert chemical names to SMILES notation using multiple chemical APIs.
+    
+    This class attempts to convert chemical compound names to SMILES notation
+    by querying multiple chemical databases APIs in parallel until a valid 
+    result is found.
+    
+    Args:
+        name: The chemical compound name to convert to SMILES notation.
+
+    Raises:
+        ValueError: If the name cannot be URL-encoded or contains invalid characters.
+    """
+
+    def __init__(self, name: str):
+        """Initialize converter with chemical name.
+        
+        Args:
+            name: Chemical compound name to convert
+            
+        Raises:
+            ValueError: If name cannot be URL-encoded
+
+        Example:
+        # Basic usage with IUPAC name
+        >>> converter = Name2Smiles("2-propanone")
+        >>> await converter.get_smiles()
+        'CC(=O)C'
+        """
+        try:
+            self.name = quote(name)
+        except Exception as e:
+            logger.error(f"Error encoding name: {e}")
+            raise ValueError(f"Invalid chemical name: {name}")
+        
+        self.timeout = 10  # seconds
+
+    async def open_molecules(self) -> Optional[str]:
+        """Query Open Molecules API for SMILES."""
+        # Implementation pending API details
+        return None
+    
+    @backoff.on_exception(backoff.expo, (aiohttp.ClientError, asyncio.TimeoutError), max_time=10, logger=logger)
+    async def opsin(self) -> Optional[str]:
```

Comment:
perhaps we could also run it locally (see the other comment). It should certainly also give more throughput.